### ISSUE TITLE
Battles having input lag for some users

### ISSUE THREAD

**Issue by panpawn** [Zarel/Pokemon-Showdown-Client#1080]
Many users have been reporting that they've been getting input lag in battles recently. After some investigation, I found that a user who was having these issues had a client crash. (Browser is Chrome, ~~but the issue is reportedly also happening on Firefox for some users, too~~ (edit, no, just Chrome))

```
>battle-gen7randombattle-706314253

|player|p2|thisisatesthehe|1
|teamsize|p1|6
|teamsize|p2|6
|gametype|singles
|gen|7
|tier|[Gen 7] Random Battle
|rated|
|seed|
|rule|Sleep Clause Mod: Limit one foe put to sleep
|rule|HP Percentage Mod: HP is shown in percentages
|
|start
|switch|p1a: Gligar|Gligar, L79, F|100/100
|switch|p2a: Sudowoodo|Sudowoodo, L83, F|252/252
|turn|1

/battle-gen7randombattle-706314253:1 Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause(). https://goo.gl/LdLk22
```

**Comment by Zarel** [comment]
I know, I used to get it too but I can't currently have the issue; I think that means the latest stable Chrome (64) has this issue but the latest beta (65) finally fixed it.

The client crash you found is unrelated, the lag caused the crash, not the other way around.

The last time I tried to figure out what was causing it, it was not anything the profiler could handle. (i.e. all of Chrome's internal performance tools were telling me Chrome was not frozen, and meanwhile the GIFs were frozen and the UI was entirely unresponsive).

Turning off animations seems to fix it, which makes me suspect the problem is 100% GIF support.

**Comment by CloudyNatu** [comment]
It'd make sense that it's GIF support, since PS! itself doesn't lag for me. Can I update Chrome to the new beta release to prevent this issue as well?

Thanks for the help!

**Comment by Zarel** [comment]
I actually can't seem to reproduce the issue anymore. Conditions causing it might be more complex than I originally thought. Updating to the new beta _should_ fix it, but unfortunately I don't have any solid data.

**Comment by Zarel** [comment]
I added a setting to turn off GIFs. In theory, a targeted fix would be to have GIFs automatically on for Chrome 64, but until I'm more sure about what's wrong, I don't want to have it on automatically _yet_.

**Comment by Zarel** [comment]
Mitigation in 8c49df9cb870502

Am still thinking about what to do about this; depends heavily on how common the freezing is.

**Comment by CloudyNatu** [comment]
Meant to respond earlier, but updating to Chrome's newest beta release does solve this issue, so I've been recommending others with the issue do that for a fix as well. Even if this issue can't be solved immediately, I'm sure Chrome 65 will be out of beta at some point in the near future, so it's a finite problem.

**Comment by Zarel** [comment]
Fixed in 6b717b8f5109
